### PR TITLE
#1221 Pending Advisor List

### LINKED ISSUE #1221: Finance - Pending Advisor List
### Description

We need the ability to generate and send a list of Sabo Ids to our advisor. this modal will serve that function from the frontend

### Acceptance Criteria

1. List should be all of the SABO #s of the requests with SABO #s assigned (ie. excluding reimbursement requests where the requester has not added a SABO #) and with status "Pending Advisor"
2. After "Sent To Advisor" is clicked, all listed reimbursement requests should have their status changed to "Approved" (chose "Approved" because this is the last step we can track ourselves and I didn't want to have something like "Pending SABO" be the final status)

### Proposed Solution

^

### Mocks

image

epic #493

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Created the Pending Advisor List Modal

## Screenshots

Screen Shot 2023-07-28 at 11 10 00 PM

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes #1221  (issue #)

### CODE REVIEW

File: src/frontend/src/apis/finance.api.ts [hunk 1]
```diff
@@ -192,3 +192,24 @@ export const downloadBlobsToPdf = async (blobData: Blob[], filename: string) =>
...
   saveAs(pdfBlob, filename);
 };
+
+/**
+ * API call to get the list of Reimbursement Requests that are pending advisor approval
+ *
+ * @returns The list of Reimbursement Requests that are pending advisor approval
+ */
+export const getPendingAdvisorList = () => {
+  return axios.get(apiUrls.financeGetPendingAdvisorList());
```
Comment (RChandler234) [inline src/frontend/src/apis/finance.api.ts:202]: call transformResponse with the frontend reimbursementRequestTransformer on all the returned reimbursement requests like we do for getAllReimbursementRequests

File: src/frontend/src/pages/FinancePage/FinancePage.tsx [hunk 2]
```diff
@@ -60,6 +76,8 @@ const FinancePage = () => {
     setAnchorEl(null);
   };
 
+  console.log(allPendingAdvisorList);
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/FinancePage.tsx:79]: delete

File: src/frontend/src/pages/FinancePage/FinancePage.tsx [hunk 3]
```diff
@@ -38,20 +43,31 @@ const FinancePage = () => {
...
-  const isFinance = user.isFinance;
+  const { isFinance, isHeadOfFinance } = user;
 
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
 
+  const [showPendingAdvisorListModal, setShowPendingAdvisorListModal] = useState(false);
+
   if (isFinance && allReimbursementRequestsIsError) return <ErrorPage message={allReimbursementRequestsError?.message} />;
   if (userReimbursementRequestIsError) return <ErrorPage message={userReimbursementRequestError?.message} />;
+  if (allPendingAdvisorListIsError) return <ErrorPage message={allPendingAdvisorListError?.message} />;
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/FinancePage.tsx:61]: isHeadOfFinance &&

File: src/frontend/src/pages/FinancePage/FinancePage.tsx [hunk 4]
```diff
@@ -101,6 +119,14 @@ const FinancePage = () => {
 
   return (
     <PageLayout title="Finance" headerRight={financeActionsDropdown}>
+      {isHeadOfFinance && (
+        <PendingAdvisorModal
+          open={showPendingAdvisorListModal}
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/FinancePage.tsx:124]: remove the && and j do isHeadOfFinance && showPendingAdvisorListModal

File: src/frontend/src/pages/FinancePage/FinancePage.tsx [hunk 5]
```diff
@@ -101,6 +119,14 @@ const FinancePage = () => {
...
+          saboNumbers={allPendingAdvisorList!.map((x) => x.saboId!)}
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/FinancePage.tsx:125]: rename x to reimbursementRequest

File: src/frontend/src/pages/FinancePage/FinanceComponents/PendingAdvisorListModal.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      await sendPendingAdvisorList(data.saboNumbers);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        toast.error(error.message);
+      }
+    }
+  };
+
+  // Custom input component for the TextField
+  const CommaSeparatedNumbersInput = ({
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/FinanceComponents/PendingAdvisorListModal.tsx:51]: you can move this component to the top of the file, and add errors as a prop. Makes it cleaner/ more reusable in other places if need be later

File: src/frontend/src/pages/FinancePage/FinanceComponents/PendingAdvisorListModal.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+  } = useForm({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      saboNumbers: saboNumbers
+    }
+  });
+
+  const toast = useToast();
+
+  if (sendPendingAdvisorListIsLoading) return <LoadingIndicator />;
```
Comment (RChandler234) [inline src/frontend/src/pages/FinancePage/FinanceComponents/PendingAdvisorListModal.tsx:38]: move this inside the modal and j disable the submit button while this is loading, it looks better

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Looks really good, just cleaning up the code and tweaking some things

File: src/frontend/src/pages/FinancePage/FinancePage.tsx (hunk 4 above)
Comment (Peyton-McKee) [reply src/frontend/src/pages/FinancePage/FinancePage.tsx:124]: huh?

Comment (Peyton-McKee) [reply src/frontend/src/pages/FinancePage/FinancePage.tsx:124]: oh

Comment (Peyton-McKee) [reply src/frontend/src/pages/FinancePage/FinancePage.tsx:124]: wait no im still confused

**Review by RChandler234 — APPROVED** [review]

**Pull request merged**